upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -150,6 +173,61 @@ func (tf *TemplateFunctions) SharedVPC() bool {
 	return tf.cluster.SharedVPC()
 }
 
+// These are the network topology functions. They are boolean logic for checking which type of
+// topology this cluster is set to be deployed with.
+func (tf *TemplateFunctions) IsTopologyPrivate()         bool  { return tf.cluster.IsTopologyPrivate() }
+func (tf *TemplateFunctions) IsTopologyPublic()          bool  { return tf.cluster.IsTopologyPublic() }
+func (tf *TemplateFunctions) IsTopologyPrivateMasters()  bool  { return tf.cluster.IsTopologyPrivateMasters() }
+
+func (tf *TemplateFunctions) WithBastion()  bool  {
+	 return !tf.cluster.Spec.Topology.BypassBastion
+}
+
+// This function is replacing existing yaml
+func (tf *TemplateFunctions) GetBastionZone() (string, error) {
+	var name string
+	if len(tf.cluster.Spec.Zones) <= 1 {
```

Comment:
Hey @kris-nova. Yeah, I ran into this when only specifying a single zone during cluster creation:
```
$ kops create cluster --topology=private --networking=cni --zones=us-west-1c  --ssh-public-key=${MYKEY} --cloud=aws "${NAME}"
I1109 22:03:08.390730    4441 cluster.go:498] Assigned CIDR 172.20.96.0/19 to zone us-west-1c
I1109 22:03:08.392186    4441 cluster.go:509] Assigned Private CIDR 172.20.128.0/19 to zone us-west-1c
I1109 22:03:09.759130    4441 populate_cluster_spec.go:237] Defaulting DNS zone to: Z1HPCW92NYZMDF
Previewing changes that will be made:


error building tasks: error handling file "cloudup/_aws/topologies/_topology_private/bastion.yaml": error executing template "_aws/topologies/_topology_private/bastion.yaml": template: _aws/topologies/_topology_private/bastion.yaml:117:24: executing "_aws/topologies/_topology_private/bastion.yaml" at <GetBastionZone>: error calling GetBastionZone: Unable to detect zone name for bastion
```

Let me know if I should create a ticket.